docs/getting_started/DMUpgrade.md

```diff
@@ -0,0 +1,310 @@
+# Upgrading the Datamodel
+
+## Purpose
+
+The purpose of Datamodel Upgrade is to make it possible to reinstall Datamodel without having to first uninstall and then reinstall it.
+
+## Parameters
+
+
+### Build parameters
+
+| Name               | Description                                   |  Type  | Value     |
+|--------------------|-----------------------------------------------|:------:|:----------|
+| `artifact_repo`    | The git repo url where all current CRDs exist | string | `git_url` |
+| `prev-spec-branch` | Branch of current CRDs to compare to new CRDs | string | `master`  |
+
+
+
+
+### Runtime/Install parameters
+
+
+| Name                                         | Description                                         | Type | Value   |
+|----------------------------------------------|-----------------------------------------------------|:----:|:--------|
+| `force`                                      | Force upgrade of a datamodel                        | bool | `false` |
+| `datamodel_backward_compatibility_validator` | Flag to enable/disable backward compatibility check | bool | `false` |
+
+Specify each parameter using the `--key=value` argument to `nexus datamodel build`. For example,
+
+```
+$ nexus datamodel build --name orgchart  --artifact_repo=<git_repo> --prev_spec_branch=<master>
+```
+
+The above command builds the new datamodel against the master branch of the artifactory repo and force will be false.<br>
+If prev_spec_branch is not provided then force will be set to true.
+
+## What is backward compatibility check?
+It is a check performed to determine the changes performed on dsl during the datamodel upgrade have backward compatibility with existing crds(source of truth).<br>
+At build time :<br>
+* if "prev-spec-branch" is not provided, force will be set to true and build/crds dir will be the source of truth if present.
+* if "prev-spec-branch" is provided and artifact_repo is provided, force will be set to false and  source of truth will be the crds dir present in prev_spec_branch of artifact_repo provided
+* if "prev-spec-branch" is provided and artifact_repo is not provided, we will check nexus.yaml for artifact_repo and if it is not present we will error and exit <br><br>
+
+At runtime crds existing in the cluster will be the source of truth 
```

Comment:
At runtime:

The existing CRDs deployed in the cluster are considered the source of truth for the backward compatibility check.